virtual_rainforest/models/abiotic/radiation.py

```diff
@@ -0,0 +1,208 @@
+"""The `abiotic.radiation` module.
+
+The radiation module calculates the radiation balance of the Virtual Rainforest.
+The top of canopy net shortwave radiation at a given location depends on
+
+1. extra-terrestrial radiation (affected by the earth's orbit, date, and location),
+
+2. terrestrial radiation (affected by atmospheric composition and clouds),
+
+3. topography (elevation, slope and aspect),
+
+4. surface albedo (vegetation type and fraction of vegetation/bare soil), and
+
+5. emitted longwave radiation.
+
+The preprocessing module takes extra-terrestrial radiation as an input and adjusts for
+the effects of topography (slope and aspect). Here, the effects of atmospheric
+filtering (elevation-dependent) and cloud cover are added to calculate photosynthetic
+photon flux density (PPFD) at the top of the canopy which is a crucial input to the
+plant module. The implementation is based on :cite:t:`Davis2017`.
+
+Cloud cover and surface albedo also determine how much of the shortwave radiation that
+reaches the top of the canopy is reflected and how much remains to be absorbed via
+photosynthesis and re-emitted as longwave radiation by vegetation and forest floor.
+Scattering and re-absorption of longwave radiation are not considered at this stage.
+
+The radiation balance is calculated with the radiation_balance() function. At the
+moment, this happens at a daily timestep.
+
+# the following structural components are not implemented yet
+TODO include time dimension
+TODO logging, raise errors
+"""
+
+# from typing import Optional
+import numpy as np
+from numpy.typing import NDArray
+
+# from core.constants import CONSTANTS as C
+# this doesn't exist yet; optional scipy
+CLOUDY_TRANSMISSIVITY = 0.25
+"""Cloudy transmittivity :cite:p:`Linacre1968`"""
+TRANSMISSIVITY_COEFFICIENT = 0.50
+"""Angular coefficient of transmittivity :cite:p:`Linacre1968`"""
+FLUX_TO_ENERGY = 2.04
+"""From flux to energy conversion, umol/J :cite:p:`Meek1984`"""
+BOLZMAN_CONSTANT = 5.67e-8
+"""Stephan Bolzman constant W m-2 K-4"""
```

Comment:
Spelling. Also, I'd refer to this as Stefan's constant to avoid confusion with the Boltzmann constant.

```suggestion
STEFANS_CONSTANT = 5.67e-8
"""Stefan-Boltzmann constant W m-2 K-4"""
```

Reply:
The correct should be is `STEFAN_BOLTZMAN_CONSTANT`, or `SIGMA`. Just `BOLTZMAN` or `STEFAN` would be misleading. https://en.wikipedia.org/wiki/Stefan–Boltzmann_constant

Reply:
I'll defer to the physicist here :wink:. Mind there are two Ns in Boltzmann though!